app/validators/answers/option_answer_validator.py

```diff
@@ -133,12 +136,25 @@ def _validate_dynamic_options_answer_source(self):
             return None
 
         value_source = self.dynamic_options["values"]
-        if (
-            value_source["source"] == "answers"
-            and self.questionnaire_schema.get_answer_type(value_source["identifier"])
-            != AnswerType.CHECKBOX
-        ):
-            self.add_error(
-                self.DYNAMIC_OPTIONS_REFERENCES_NON_CHECKBOX_ANSWER,
-                value_source=value_source,
-            )
+        transform = self.dynamic_options["transform"]
+
+        if value_source["source"] == "answers":
+
+            if (
+                self.questionnaire_schema.get_answer_type(value_source["identifier"])
+                != AnswerType.CHECKBOX
+            ):
+
+                self.add_error(
+                    self.DYNAMIC_OPTIONS_REFERENCES_NON_CHECKBOX_ANSWER,
+                    value_source=value_source,
+                )
+
+            if (
+                "option-label-from-value" in transform
+                and transform["option-label-from-value"][1]
+                != value_source["identifier"]
+            ):
+                self.add_error(
+                    self.DYNAMIC_OPTIONS_SOURCE_IDENTIFIER_AND_OPTION_LABEL_FROM_VALUE_MISMATCH
+                )
```

Comment:
We should look to add context where possible. This would benefit from source/transform identifiers.
```suggestion
                self.add_error(
                    self.DYNAMIC_OPTIONS_SOURCE_IDENTIFIER_AND_OPTION_LABEL_FROM_VALUE_MISMATCH,
                    source_identifier=value_source["identifier"],
                    transform_identifier=transform["option-label-from-value"][1]
                )
```

Reply:
Thought I would treat myself to a github commit, never use them usually. Linting and test failures 😆